Fb376

joelaro04 opened the pull request:
AugurClient would only watch for `OrderCreated` events and grab the associated `market_id`. This would be okay in our case, since we would only need to populate the top markets for now, but we can add other events to watch as well.

joelaro04 committed:
Merge branch 'master' of github.com:spiretrading/pyaugur

joelaro04 committed:
Merge branch 'master' of github.com:spiretrading/pyaugur

joelaro04 committed:
Added listeners for Augur events.

Added AugurClient to __init__.py

Added unittest for AugurClient. Added run_command method

Added ethereum client to AugurClient.

Added start_augur_event_listener method.

Added map_event_name_to_signature method.

Able to obtain Augur events and which market_id it belongs to.

Removed methods that Seenex should implement.
- Moved methods not needing self outside of AugurClient.

Modified start_augur_event_listener to also retrieve events since last poll.

Added start_augur_event_historical_listener.

Added start_augur_top_market_listener.

Added event_name to callback.

Removed unused methods.

Removed fetch_history variable

Changed listeners to accept event_names.
- Generate event signature map on initialization.

kranar commented on augur/augur_client.py:
```diff
@@ -44,7 +47,7 @@ def _ensure_outcome_info(object_dict):
   '''Client for connecting and interacting with an Augur Node.'''
 
-  def __init__(self, hostname, port):
+  def __init__(self, hostname, port, ethereum=None):
```
Use a more descriptive parameter name. Document all parameters.

kranar commented on augur/augur_client.py:
```diff
@@ -53,8 +56,10 @@ def __init__(self, hostname, port):
     self._hostname = hostname
     self._port = port
-    self._node = None
+    self._augur = None
```
Similarly, a variable named ```_augur``` gives no information about what this is. Augur is an entire decentralized prediction platform so calling a variable by that name gives a reader no context.

kranar commented on augur/augur_client.py:
```diff
@@ -120,15 +125,89 @@ def __init__(self, hostname, port):
       ([_ensure_outcome_info(outcome)
         for outcome in data['outcomes'] if outcome]))
 
+  async def start_augur_event_historical_listener(self, from_block, on_event):
```
Remove the ```augur``` from the name. This entire client only relates to augur.

kranar commented on augur/augur_client.py:
```diff
@@ -120,15 +125,89 @@ def __init__(self, hostname, port):
+      self._augur = await websockets.connect('ws://{}:{}'.format(
+        self._hostname, str(self._port)))
+      await self._load_sync_data()
+      abi_url = os.environ['AUGUR_CONTRACT_ABI']
```
Don't use environment variables.

joelaro04 committed:
Added methods to retrieve market id from an event.

Added block listener

Added public method documentation

Removed underscores from functions.

Removed print statements and unused imports.

Added test cases for loading market id from hash.
